mne/utils/check.py

```diff
@@ -525,6 +526,7 @@ def __instancecheck__(cls, other):
     "path-like": path_like,
     "int-like": (int_like,),
     "callable": (_Callable(),),
+    "array-like": (Sequence, np.ndarray),
```

Comment:
Have you tried using `numpy.typing.ArrayLike`?

Reply:
Never mind, this one would also accept a scalar, so it's not suitable:

https://numpy.org/devdocs/reference/typing.html#numpy.typing.ArrayLike

Reply:
It's also a `Union`, so we can't use it directly as a type in our little dict. OK to keep it as is?

Reply:
> It's also a `Union`, so we can't use it directly as a type in our little dict.

I don't understand that, but I'm okay with keeping it the way you proposed, yes

Reply:
I can't just add `"array-like": (np.typing.ArrayLike)` instead of `"array-like": (Sequence, np.ndarray)` to our type dict, because it produces a weird error. I don't understand it in detail either, and I don't even want to check because Python typing is weird 😄.